components/form-builder/plate-editor/helpers/markdown.ts

```diff
@@ -1,57 +1,38 @@
-import {
-  ELEMENT_PARAGRAPH,
-  ELEMENT_BLOCKQUOTE,
-  ELEMENT_LINK,
-  MARK_CODE,
-  MARK_ITALIC,
-  MARK_BOLD,
-  MARK_STRIKETHROUGH,
-  ELEMENT_IMAGE,
-  ELEMENT_CODE_BLOCK,
-  ELEMENT_HR,
-  ELEMENT_UL,
-  ELEMENT_OL,
-  ELEMENT_LI,
-  ELEMENT_H1,
-  ELEMENT_H2,
-  ELEMENT_H3,
-  ELEMENT_H4,
-  ELEMENT_H5,
-  ELEMENT_H6,
-} from "@udecode/plate";
 import serialize from "./serialize";
+import { BlockType, LeafType } from "remark-slate";
 
-export const serializeMd = (value) => {
-  return serialize(
-    { children: value },
-    {
-      nodeTypes: {
-        paragraph: ELEMENT_PARAGRAPH,
-        block_quote: ELEMENT_BLOCKQUOTE,
-        link: ELEMENT_LINK,
-        inline_code_mark: MARK_CODE,
-        emphasis_mark: MARK_ITALIC,
-        strong_mark: MARK_BOLD,
-        delete_mark: MARK_STRIKETHROUGH,
-        // NOTE: underline, subscript superscript not yet supported by remark-slate
-        // underline: MARK_UNDERLINE,
-        // subscript: MARK_SUBSCRIPT,
-        // superscript: MARK_SUPERSCRIPT,
-        image: ELEMENT_IMAGE,
-        code_block: ELEMENT_CODE_BLOCK,
-        thematic_break: ELEMENT_HR,
-        ul_list: ELEMENT_UL,
-        ol_list: ELEMENT_OL,
-        listItem: ELEMENT_LI,
-        heading: {
-          1: ELEMENT_H1,
-          2: ELEMENT_H2,
-          4: ELEMENT_H4,
-          3: ELEMENT_H3,
-          5: ELEMENT_H5,
-          6: ELEMENT_H6,
-        },
+export const serializeMd = (value: (BlockType | LeafType)[]) => {
```

Comment:
@dsamojlenko 

Think I tracked down the TS errors on this.

<hr>

I console logged out the value coming into `serializeMd `

From what I can see it's an array with a single element so value[0] which contains either block nodes or leaf nodes

```
 export const serializeMd = (value: (BlockType | LeafType)[]) => {
```

The code you had 

```
return serialize(
    { children: value },
```

Was kind of forcing it to look like a block as Leafs don't have children... at which point it was complaining that the "block" was missing a type prop.

Value[0]  (if not undefined seems to fixes all that) --- seems to work from my testing

------

After that TS it started complaining that a "p" was not assignable to paragraph for the NodeType opts

TypeScript autocompleted those as just strings so I ended up replacing the node types with the string equivalents

Reply:
... further updates for the onChange now use the Value type  ->TDescendant = TElement | TText;

Which pretty my maps back to BlockType | LeafType

Reply:
Found this issue in the remark-slate repo: https://github.com/hanford/remark-slate/issues/57

...that explains the issue with the nodeTypes. I updated the serialize script to use the custom nodeTypes required by Plate: https://github.com/cds-snc/platform-forms-client/pull/1059/commits/c00f56377d1544ffb926391170d79f37af770f28